docs/pages/deploy-a-cluster/high-availability.mdx

```diff
@@ -0,0 +1,520 @@
+  credential provisioner will need to manage DNS records to demonstrate control
+  over your domain name.
+
+![Diagram of a high-availability Teleport
+architecture](../../img/deploy-a-cluster/teleport-ha-architecture.png)
+
+## Layer 4 load balancer
+
+The load balancer forwards traffic from users and services to an available
+Teleport instance. This must not terminate TLS, and must transparently forward
+the TCP traffic it receives. In other words, this must be a Layer 4 load
+balancer, not a Layer 7 (e.g., HTTP) load balancer. 
+
+### TLS Routing
+
+Your load balancer configuration depends on whether you will enable [TLS
+Routing](../management/operations/tls-routing.mdx) in your Teleport cluster.
+
+With TLS Routing, the Teleport Proxy Service uses application-layer protocol
+negotiation (ALPN) to handle all communication with users and services via the
+HTTPS port, regardless of protocol. Without TLS Routing, the Proxy Service
+listens on a separate port for each protocol.
+
+The advantage of TLS Routing is its simplicity: you can expose only a single
+port on the load balancer to the public internet.
+
+The disadvantage of TLS Routing is that it is impossible to implement Layer 7
+load balancing for HTTPS traffic, since traffic that reaches the HTTPS port can
+use any supported protocol. 
+
+The approach we describe in this guide uses only a Layer 4 load balancer to
+minimize the infrastructure you will deploy, but users that require a separate
+load balancer for HTTPS traffic should disable TLS Routing.
+
+### Configuring the load balancer
+
+Configure the load balancer to forward traffic from the following ports on the
+load balancer to the corresponding port on an available Teleport instance. The
+configuration depends on whether you will enable TLS Routing:
+
+<Tabs>
+<TabItem label="TLS Routing">
+
+| Port | Description |
+| - | - |
+| `443` | ALPN port for TLS Routing. |
+
+</TabItem>
+<TabItem label="Separate Ports">
+
+These ports are required:
+
+| Port | Description |
+| - | - |
+| `3023` | SSH port for clients connect to. |
+| `3024` | SSH port used to create reverse SSH tunnels from behind-firewall environments. |
+| `443` | HTTPS connections to authenticate `tsh` users into the cluster. The same connection is used to serve a Web UI. |
+
+You can leave these ports closed if you are not using their corresponding
+services:
+
+| Port | Description |
+| - | - |
+| `3026` | HTTPS Kubernetes proxy |
+| `3036` | MySQL port |
+| `5432` | Postgres port |
+
+</TabItem>
+</Tabs>
+
+## Key-value store
+
+The Teleport Auth Service stores cluster state (such as dynamic configuration
+resources) and audit events as key/value pairs. In high-availability
+deployments, you must configure the Auth Service to manage this data in a
+key-value store that runs outside of your cluster of Teleport instances.
+
+The Auth Service supports the following backends for cluster state and audit
+events:
+
+- Amazon DynamoDB
+- Google Cloud Firestore
```

Comment:
I see etcd mentioned below, but why is it not included in this list?

Reply:
I removed it since this is a list of solutions we support for cluster state and audit events, and we don't support etcd for audit events (https://github.com/gravitational/teleport/pull/19217#discussion_r1044817411). It does make sense to put etcd in this list, though, so I'll add it with a parenthetical to indicate we don't support it for audit events.